Bucketeer/Sources/Public/BKTClient.swift

```diff
@@ -126,24 +143,50 @@ extension BKTClient {
         }
     }
 
-    public func stringVariation(featureId: String, defaultValue: String) -> String {
-        return getVariationValue(featureId: featureId, defaultValue: defaultValue)
+    public func boolVariation(featureId: String, defaultValue: Bool) -> Bool {
+        return boolVariationDetails(featureId: featureId, defaultValue: defaultValue).variationValue
+    }
+
+    public func boolVariationDetails(featureId: String, defaultValue: Bool) -> BKTEvaluationDetails<Bool> {
+        return getVariationDetail(featureId: featureId, defaultValue: defaultValue)
     }
 
     public func intVariation(featureId: String, defaultValue: Int) -> Int {
-        return getVariationValue(featureId: featureId, defaultValue: defaultValue)
+        return intVariationDetails(featureId: featureId, defaultValue: defaultValue).variationValue
+    }
+
+    public func intVariationDetails(featureId: String, defaultValue: Int) -> BKTEvaluationDetails<Int> {
+        return getVariationDetail(featureId: featureId, defaultValue: defaultValue)
     }
 
     public func doubleVariation(featureId: String, defaultValue: Double) -> Double {
-        return getVariationValue(featureId: featureId, defaultValue: defaultValue)
+        return doubleVariationDetails(featureId: featureId, defaultValue: defaultValue).variationValue
     }
 
-    public func boolVariation(featureId: String, defaultValue: Bool) -> Bool {
-        return getVariationValue(featureId: featureId, defaultValue: defaultValue)
+    public func doubleVariationDetails(featureId: String, defaultValue: Double) -> BKTEvaluationDetails<Double> {
+        return getVariationDetail(featureId: featureId, defaultValue: defaultValue)
+    }
+
+    public func stringVariation(featureId: String, defaultValue: String) -> String {
+        return stringVariationDetails(featureId: featureId, defaultValue: defaultValue).variationValue
+    }
+
+    public func stringVariationDetails(featureId: String, defaultValue: String) -> BKTEvaluationDetails<String> {
+        return getVariationDetail(featureId: featureId, defaultValue: defaultValue)
+    }
```

Comment:
Can we call the same function for both cases?
It's a bit confusing. The non-details functions call the `stringVariationDetails` but the details function calls a different function `getVariationDetail`.

E.g.
```suggestion
    public func stringVariation(featureId: String, defaultValue: String) -> String {
        return stringVariationDetails(featureId: featureId, defaultValue: defaultValue).variationValue
    }

    public func stringVariationDetails(featureId: String, defaultValue: String) -> BKTEvaluationDetails<String> {
        return stringVariationDetails(featureId: featureId, defaultValue: defaultValue)
    }
```

Reply:
Like this.
https://github.com/bucketeer-io/go-server-sdk/pull/136/files#diff-9658208f158246a9cfbc2c798e77e8224eaed053c42d140eb1e7237d8ed4e91cR295-R305